File: App/Questionnaire/src/main/java/com/example/mentalhealth/aiService/DTO/ComponentInsightDTO.java
package com.example.mentalhealth.aiService.DTO;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;
@Data
@AllArgsConstructor
@NoArgsConstructor
public class ComponentInsightDTO {

    private String componentId;              // comp2_sleep_latency
    private String componentName;            // human readable
    private Double score;                    // component score
    private String interpretationHint;       // OPTIONAL short phrase
    private List<String> contributingQuestionIds;
}

File: App/AI-service/models/ComponentInsightDTO.py
from pydantic import BaseModel, Field
from typing import List, Optional


class ComponentInsightDTO(BaseModel):
    componentId: str = Field(..., description="Component identifier")

    componentName: str = Field(
        ..., description="Human readable component name"
    )

    score: float = Field(..., description="Component score")

    interpretationHint: Optional[str] = Field(
        None,
        description="Short hint about component interpretation"
    )

    contributingQuestionIds: Optional[List[str]] = []

    class Config:
        extra = "forbid"